cmd/influx/cli/cli.go

```diff
@@ -73,6 +77,22 @@ func New(version string) *CommandLine {
 
 // Run executes the CLI
 func (c *CommandLine) Run() error {
+	// If we are not running in an interactive terminal, read stdin completely
+	// and execute a query. Do not allow meta commands.
+	if !c.ForceTTY && !terminal.IsTerminal(int(os.Stdin.Fd())) {
+		if err := c.Connect(""); err != nil {
+			return fmt.Errorf(
+				"Failed to connect to %s\nPlease check your connection settings and ensure 'influxd' is running.",
+				c.Client.Addr())
+		}
+
+		cmd, err := ioutil.ReadAll(os.Stdin)
```

Comment:
Is it safe to read all of stdin for this? Prior to this change, stdin would have been read one line at a time, AFAICT. Should this use a bufio.Scanner instead, to handle the input line-by-line?